server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -317,18 +325,49 @@ public IndexInput openInput(String name, IOContext context) throws IOException {
         }
     }
 
-    public void copyFrom(Directory from, String src, String dest, IOContext context, boolean useCommonSuffix) throws IOException {
+    public void acquireLock(long primaryTerm, long generation, String resourceID) throws IOException {
+        Optional<String> metadataFile = getMetadataFileForCommit(primaryTerm, generation);
+        if (metadataFile.isEmpty()) {
+            String errorString = "Metadata file is not present for given primary term "
+                + primaryTerm + " and generation " + generation;
+            throw new FileNotFoundException(errorString);
+        }
+        mdLockManager.acquire(mdLockManager.getLockInfoBuilder().withMetadataFile(metadataFile.get())
+            .withResourceId(resourceID).build());
+    }
+
+    // Visible for testing
+    Optional<String> getMetadataFileForCommit(long primaryTerm, long generation) throws IOException {
+        Collection<String> metadataFiles = remoteMetadataDirectory.listFilesByPrefix(
+            MetadataFilenameUtils.METADATA_PREFIX);
```

Comment:
If we add `primaryTerm` and `generation` in the argument, it will help in fetching fewer bytes over the wire.